python/qa_rest_api.py

```diff
@@ -53,89 +52,31 @@ def test_get_status(self):
             self.assertEqual(check_str in dict_str, True)
         self.assertEqual('fake_var' in dict_str, False)
 
-    def test_get_config(self):
-        self.pi_val = 3.14
-        self.pi_copy = self.pi_val
+    def test_config_by_name(self):
+        self.var1 = None
+        self.var2 = None
         port = self.get_free_port()
-        instance = RestApi(self, port, write_settings=['pi_val'])
+        instance = RestApi(self, port, write_settings=['var1', 'var2'])
         # give server a second to startup
         time.sleep(1)
+        self.var1 = 0
+        self.var2 = 0
         addr_str = 'http://127.0.0.1:' + str(port) + '/config'
-        resp = httpx.Client().get(addr_str)
+        resp = httpx.Client().put(addr_str, json={"var1": 3.14, "var2": 5.0})
         self.assertIsNotNone(resp)
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
-        dict_str = resp.content.decode("UTF-8")
-        res_dict = ast.literal_eval(dict_str)
-        self.assertEqual(res_dict['pi_val'], self.pi_val)
-        self.assertEqual('pi_copy' in dict_str, False)
+        self.assertEqual(self.var1, 3.14)
+        self.assertEqual(self.var2, 5.0)
 
-    def test_put_config(self):
-        self.pi_val = 0
-        self.test_int = 1
+    def test_bad_config_by_name(self):
+        self.var1 = None
         port = self.get_free_port()
-        instance = RestApi(self, port, read_settings=['pi_val', 'test_int'], write_settings=['pi_val'])
+        instance = RestApi(self, port, write_settings=['var1'])
         # give server a second to startup
         time.sleep(1)
+        self.var1 = 0
         addr_str = 'http://127.0.0.1:' + str(port) + '/config'
-        resp = httpx.Client().put(addr_str, json={"pi_val": 3.14})
-        self.assertIsNotNone(resp)
-        self.assertEqual(resp.status_code, status.HTTP_200_OK)
-        dict_str = resp.content.decode("UTF-8")
-        res_dict = ast.literal_eval(dict_str)
-        self.assertEqual(res_dict['pi_val'], 3.14)
-
-    def test_call_by_name(self):
-        self.val = None
-        port = self.get_free_port()
-        instance = RestApi(self, port, write_settings=['update_var', 'update_var1'])
-        # give server a second to startup
-        time.sleep(1)
-        self.val = 0
-        self.val1 = 0
-        addr_str = 'http://127.0.0.1:' + str(port) + '/call'
-        resp = httpx.Client().put(addr_str, json={"update_var": 3.14, "update_var1": 5.0})
-        self.assertIsNotNone(resp)
-        self.assertEqual(resp.status_code, status.HTTP_200_OK)
-        self.assertEqual(self.val, 3.14)
-        self.assertEqual(self.val1, 5.0)
-
-    def test_unauthorized_put_config(self):
```

Comment:
any particular reason why we deleted this test?

Reply:
it doesnt support those calls anymore